Documentation/postmortems/v3.5-data-inconsistency.md

```diff
@@ -0,0 +1,121 @@
+
+In cluster with multiple members it would mean that one of the members that crashed, will missing changes from failed apply workflow.
+This means that it will have different state of database and will return different hash via `HashKV` grpc call.
+
+There is an automatic mechanism to detect data inconsistency. 
+It can be executed during etcd start via `--experimental-initial-corrupt-check` and periodically via `--experimental-corrupt-check-time`.
+Both checks however have a flaw, they depend on `HashKV` grpc method, which might fail causing the check to pass.
+
+In multi member etcd cluster, each member can run with different performance and be at different stage of applying the WAL log.
+Comparing database hashes between multiple etcd members requires all hashes to be calculated at the same change.
+This is done by requesting hash for the same `revision` (version of key value store). 
+However, it will not work if the provided revision is not available on the members.
+This can happen on very slow members, or in cases where corruption has lead revision numbers to diverge.
+
+This means that for this issue, the corrupt check is only reliable during etcd start just after etcd the crash.
+
+## Impact
+
+We are not aware any cases of users reporting a data corruption in production environment.
+
+However, issue was critical enough to motivate a public statement. 
+Main impact comes from loosing user trust into etcd reliability.
+
+## Lessons Learned
+
+### What went well
+
+* Multiple maintainers were able to work effectively on reproducing and fixing the issue. As they are in different timezones, there was always someone working on the issue.
+
+### What went wrong
+
+* TODO: Etcd code is over complicated
+* No users enable data corruption detection as it is still an experimental feature introduced in v3.3. All reported cases where detected manually making it almost impossible to reproduce.
+* Etcd has functional tests designed to detect such problems, however they are unmaintained, flaky and are missing crucial scenarios.
+* Etcd v3.5 release was not qualified as well as previous ones. Older maintainers run manual qualification process that is no longer known or executed.
+* TODO: Fixing the data inconsistency went poorly
+
+### Where we got lucky
+
+* We reproduced the issue using etcd functional only because weird partition setup on workstation. Functional tests store etcd data under `/tmp` usually mounted to in memory filesystem. Problem was reproduced only because one of the maintainers has `/tmp` mounted to standard disk.
+
+## Action items
+
+| Action Item                                                                         | Type    | Owner | Bug |
+|-------------------------------------------------------------------------------------|---------|-------|-----|
+| Data corruption detection is enabled by default                                     | Detect  |       |     |
```

Comment:
May be we should merge this and the next item into one for better tracking later on? (i.e. the 1st item is read as a sub-item of the next one)